modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/MustacheScriptEngineService.java

```diff
@@ -85,8 +88,21 @@ public MustacheScriptEngineService(Settings settings) {
      * */
     @Override
     public Object compile(String template, Map<String, String> params) {
-        /** Factory to generate Mustache objects from. */
-        return (new JsonEscapingMustacheFactory()).compile(new FastStringReader(template), "query-template");
+        boolean escapeJson = true;
+        String escapeJsonStr = params.get(ESCAPE_JSON_PARAM);
```

Comment:
The main reason I chose the `escape_json` and default to true approach is for isolating this change just to the `lang-mustache` module. Changing this to be based on content type and defaulting to be plain text, would mean that this change gets larger, while the default for ES is escaping json. Also currently there is only json escaping or no escaping. 

I'm not against this approach, so I'll try this out and see how much the impact is.